prov/gni/src/gnix_av.c

```diff
@@ -155,40 +157,51 @@ static int table_insert(struct gnix_fid_av *av_priv, const void *addr,
 			size_t count, fi_addr_t *fi_addr, uint64_t flags,
 			void *context)
 {
-	struct gnix_ep_name *temp = NULL;
+	struct gnix_ep_name ep_name;
 	int ret = count;
-	size_t index;
-	size_t i;
+	size_t index, i;
 	int *entry_err = context;
+	void (*get_ep_name)(const char *, int, struct gnix_ep_name *);
 
 	if (gnix_check_capacity(av_priv, count)) {
 		return -FI_ENOMEM;
 	}
 
 	assert(av_priv->table);
+
+	/* Use a function pointer to resolve the address */
+	if (av_priv->domain->addr_format == FI_ADDR_STR) {
+		get_ep_name = _gnix_resolve_str_ep_name;
+	} else {
+		get_ep_name = _gnix_resolve_gni_ep_name;
+	}
```

Comment:
I don't like the function pointer either. There isn't any proof that this helps KNL, infact KNL has a smaller branching cache(I can't recall the exact name here), meaning fewer functions are better. If anything, it would benefit from the use of static inline functions more than a function pointer which it must resolve.